include/swift/AST/DiagnosticBridge.h

```diff
@@ -0,0 +1,68 @@
+//===--- DiagnosticBridge.h - Diagnostic Bridge to SwiftSyntax --*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  This file declares the DiagnosticBridge class, which bridges to swift-syntax
+//  for diagnostics printing.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_DIAGNOSTICBRIDGE_H
+#define SWIFT_BASIC_DIAGNOSTICBRIDGE_H
+
+#include "swift/AST/DiagnosticConsumer.h"
+#include "swift/Basic/SourceManager.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace swift {
+class DiagnosticBridge {
```

Comment:
Nit: might be good to document that all the methods are only defined when `SWIFT_BUILD_SWIFT_SYNTAX` is enabled; otherwise I think you get a linking error.